COM-614 Add testing indicator 5 function

lauravignoli opened the pull request:
This is used for all parts except Confirmed through Viral Load Testing Disaggregation.

lauravignoli committed:
COM-614 Add testing indicator 5 function

This is used for all parts except Confirmed through Viral Load Testing Disaggregation.

tmvumbi2 commented on reportssql/testing_report_functions.sql:
```diff
@@ -269,6 +269,39 @@ WHERE
+        patient pat
+    WHERE
+        patientGenderIs(pat.patient_id, p_gender) AND
+        (previousHIVTestDateFromCounselingForm  < TIMESTAMPADD(MONTH, -1, CURDATE()) OR previousHIVTestDateFromCounselingForm IS NULL) AND
```
@lauravignoli the last HIV test date needs to be less than a month ago ; the first part of the condition should be `previousHIVTestDateFromCounselingForm  > TIMESTAMPADD(MONTH, -1, CURDATE())`. Please see test case 3 for Indicator 5a

lauravignoli committed:
COM-634 Fix testing indicator 5 function

Previous HIV test date should be less than a month old, and was previously checking for older than a month.

tmvumbi2 committed:
COM-614 Fix bugs in functions for testing indicator 1 and 5
The previous code assumed that the variable previousHIVTestDateFromCounselingForm is updated for each row being selected, but it turns out this is not how `SELECT col INTO variable `works; the variable is updated only once at the end of the execution of the entire query. The bug fix consists of removing the variable previousHIVTestDateFromCounselingForm and calling the function directly in the where clause.
